File: python/190_Reverse Bits.py
# 190. Reverse Bits
class Solution(object):
    def reverseBits(self, n):
        """
        :type n: int
        :rtype: int
        """
        ans = 0
        for i in xrange(32):
            ans = (ans << 1) + (n & 1)
            n >>= 1
        return ans

File: java/190_Reverse_Bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        for (int i = 0; i < 16; i++) {
            if ((n >> i & 1) != (n >> (31 - i) & 1)){
                n ^= 1 << i ^ 1 << (31 - i);
            }
        }
        return n;
    }
}